crates/bevy_ecs/src/reflect/bundle.rs

```diff
@@ -35,7 +35,8 @@ pub struct ReflectBundleFns {
     /// Function pointer implementing [`ReflectBundle::apply_or_insert()`].
     pub apply_or_insert: fn(&mut EntityWorldMut, &dyn PartialReflect, &TypeRegistry),
     /// Function pointer implementing [`ReflectBundle::remove()`].
-    pub remove: fn(&mut EntityWorldMut),
+    pub remove: fn(&mut EntityWorldMut) -> ReflectBundle,
+    pub take: fn(&mut EntityWorldMut) -> Option<ReflectBundle>,
```

Comment:
Imo `take` should return the bundle as a `Box<dyn Reflect>`

Reply:
I think this might also be a confusion of concepts (apologies if untrue)? A `ReflectBundle` by itself holds no runtime information. It's all just type information. So to insert the bundle you need to pair it with a matching `dyn Reflect`. Likewise, to take the bundle you need to return it as a `dyn Reflect`.

Hopefully that helps clear things up!